lib/xlsxtream/workbook.rb

```diff
@@ -62,6 +62,14 @@ def initialize(output, options = {})
     end
 
     def add_worksheet(*args)
+      if block_given?
+        fail Error, "Call `write_worksheet` to pass a block"
+      end
```

Comment:
Do we really need this? It would require a major version bump, since we break backwards compatibility.

Reply:
Good point, I'll drop it

Reply:
I think you need to list `&block` as an argument to the add_worksheet method, or you will get a NameError (`def add_worksheet, *args, &block)`.

Reply:
Fixed in dbfe5e91ea8fbfa1501e536baab6f6870d8c83b4.